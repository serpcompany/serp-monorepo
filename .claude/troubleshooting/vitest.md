https://vitest.dev/guide/common-errors.html