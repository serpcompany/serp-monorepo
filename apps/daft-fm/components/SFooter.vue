<template>
  <footer class="mt-10 py-10">
    <div class="border-top container mx-auto px-4">
      <!-- Main Footer Navigation -->
      <nav
        aria-label="Footer Main Navigation"
        class="mb-12 grid grid-cols-1 gap-8 md:grid-cols-3 lg:grid-cols-5"
      >
        <div v-for="column in footerColumns" :key="column.slug">
          <span class="text-md font-extrabold uppercase">
            {{ column.title }}
          </span>
          <ul class="space-y-2 pt-4" :aria-label="`${column.title} menu`">
            <li v-for="item in column.items" :key="item.slug">
              <nuxt-link :to="item.slug" class="text-s">
                {{ item.text }}
              </nuxt-link>
            </li>
          </ul>
        </div>
      </nav>

      <!-- Bottom Footer -->
      <div class="border-gray-200 pt-8">
        <!-- Logo -->
        <div class="mb-8 flex items-center justify-between">
          <div class="flex items-center space-x-2">
            <span class="text-2xl font-bold">{{ companyName }}</span>
          </div>
        </div>

        <div
          class="mb-8 flex flex-col items-center justify-between md:flex-row"
        >
          <!-- Social Links -->
          <nav aria-label="Social Media Links" class="mb-4 flex md:mb-0">
            <nuxt-link
              v-for="social in socialLinks"
              :key="social.name"
              :to="social.link"
              :aria-label="social.name"
              class="mr-6"
              target="_blank"
            >
              <u-icon
                :name="social.icon"
                class="size-5 text-white"
                aria-hidden="true"
              />
              <span class="sr-only">Visit our {{ social.name }} page</span>
            </nuxt-link>
          </nav>

          <!-- Brand links -->
          <nav
            aria-label="Brand Navigation"
            class="flex flex-wrap justify-center gap-4 text-sm md:justify-end"
          >
            <nuxt-link
              v-for="link in brandLinks"
              :key="link.slug"
              :to="link.slug"
            >
              {{ link.name }}
            </nuxt-link>
          </nav>
        </div>

        <!-- Legal links -->
        <nav
          aria-label="Legal Navigation"
          class="flex flex-col items-center justify-between text-sm md:flex-row"
        >
          <div class="mb-2 md:mb-0">
            {{ copyrightText }}
          </div>
          <div class="flex flex-wrap justify-center gap-4 md:justify-end">
            <nuxt-link
              v-for="link in legalLinks"
              :key="link.text"
              :href="link.url"
              >{{ link.text }}</nuxt-link
            >
          </div>
        </nav>

        <!-- Address -->
        <address class="mt-2 text-sm">
          {{ address }}
        </address>
      </div>
    </div>
  </footer>
</template>

<script setup lang="ts">
const config = useRuntimeConfig();
const companyName = config.public.siteName;
const socialLinks = config.public.socialLinks;
const brandLinks = config.public.brandLinks;
const copyrightText = config.public.copyrightText;
const address = config.public.address;

const legalLinks = [
  { text: 'Privacy', url: '/legal/privacy-policy' },
  { text: 'Terms', url: '/legal/terms-conditions' },
  { text: 'Affiliate Disclosure', url: '/legal/affiliate-disclosure' },
  { text: 'DMCA', url: '/legal/dmca' }
];

const footerColumns = [
  {
    title: 'Music',
    slug: '',
    items: [
      { text: 'Artists', slug: '/artists' },
      { text: 'Albums', slug: '/albums' },
      { text: 'Songs', slug: '/songs' }
    ]
  }
  // {
  //   title: 'Placeholder',
  //   slug: 'Placeholder',
  //   items: [{ text: 'Placeholder', slug: '#' }]
  // },
  // {
  //   title: 'Placeholder',
  //   slug: '#',
  //   items: [{ text: 'Placeholder', slug: '#' }]
  // },
  // {
  //   title: 'Placeholder',
  //   slug: '#',
  //   items: [
  //     { text: 'Placeholder', slug: '#' },
  //     { text: 'Placeholder', slug: '#' }
  //   ]
  // },
  // {
  //   title: 'Placeholder',
  //   slug: '#',
  //   items: [{ text: 'Placeholder', slug: '#' }]
  // }
];
</script>
