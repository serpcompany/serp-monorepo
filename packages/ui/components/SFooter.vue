<script setup lang="ts">
  import type { FooterColumn } from '@serp/types/types';

  const config = useRuntimeConfig();
  const appConfig = useAppConfig();

  const companyName = config.public.siteName;

  const socialLinks =
    appConfig.site?.socialLinks ||
    (config.public.socialLinks as Array<{
      name: string;
      href: string;
      icon: string;
    }>);

  const legalLinks =
    appConfig.site?.legalLinks ||
    (config.public.legalLinks as Array<{
      text: string;
      slug: string;
    }>);

  const footerColumns =
    appConfig.site?.footerColumns ||
    (config.public.footerColumns as FooterColumn[]);

  const footerColumnsData = computed(() => {
    const columns = footerColumns;
    return columns.map((column) => ({
      label: column.title,
      children: column.items.map((item) => ({
        label: item.text || item.name,
        to: item.slug || item.href,
      })),
    }));
  });
</script>

<template>
  <UFooter
    class="footer-wrapper bg-primary-800 dark:bg-primary-950 mt-20 w-full"
    :ui="{
      top: 'py-0 lg:py-0',
      center: 'py-0 lg:py-0',
      bottom: 'py-0 lg:py-0',
    }"
  >
    <!-- Top section with columns - full width with background -->
    <template #top>
      <div class="w-full py-30">
        <div class="mx-auto max-w-[1248px] px-4">
          <UFooterColumns
            :columns="footerColumnsData"
            :ui="{
              link: 'text-primary-100  hover:text-primary-200 dark:text-primary-100 dark:hover:text-primary-200',
              label: 'text-primary-100 text-xl dark:text-primary-100',
            }"
          >
            <!-- Left section with newsletter -->
            <template #left>
              <div class="flex flex-col items-start space-x-2">
                <span
                  class="text-primary-100 dark:text-primary-100 pb-4 text-4xl font-bold"
                >
                  {{ companyName }}
                </span>
                <p class="text-primary-100 dark:text-primary-100 font-bold">
                  Subscribe to the newsletter.
                </p>
                <p class="text-primary-100 dark:text-primary-100">
                  Join ~trillions of informed readers.
                </p>
                <NewsletterSignup />
              </div>
            </template>
          </UFooterColumns>
        </div>
      </div>
    </template>

    <!-- Bottom section with full-width colored ribbon -->
    <template #bottom>
      <div
        class="mx-auto flex max-w-[1248px] flex-col items-start justify-between px-4 pb-8 md:flex-row md:items-center"
      >
        <!-- <div class="flex flex-row items-center md:mb-0"> -->
        <!-- Social Links -->
        <div aria-label="Social Media Links" class="mb-8 flex md:mb-0">
          <NuxtLink
            v-for="social in socialLinks"
            :key="social.name"
            :to="social.href"
            :aria-label="social.name"
            class="mr-6"
            target="_blank"
          >
            <UIcon
              :name="social.icon"
              class="text-primary-100 dark:text-primary-100 size-5"
              aria-hidden="true"
            />
            <span class="sr-only">Visit our {{ social.name }} page</span>
          </NuxtLink>
        </div>
        <!-- </div> -->
        <div
          aria-label="Legal Navigation"
          class="flex flex-col text-sm sm:items-center md:flex-row"
        >
          <div
            class="flex flex-col flex-wrap justify-center gap-4 sm:flex-row md:justify-end"
          >
            <NuxtLink
              v-for="link in legalLinks"
              :key="link.text"
              :to="link.slug"
              class="text-primary-100 dark:text-primary-100"
            >
              {{ link.text }}
            </NuxtLink>
          </div>
        </div>
      </div>
    </template>
  </UFooter>
</template>
