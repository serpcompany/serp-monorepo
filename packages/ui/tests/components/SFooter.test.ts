import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import SFooter from '../../components/SFooter.vue';
import ComponentRender from '../componentRender';

// Initialize a default runtime config object
let config_: Record<string, unknown> = {
  app: { baseURL: '/' },
  public: {
    siteName: 'Default Company',
    socialLinks: [],
    brandLinks: [],
    copyrightText: '',
    address: '',
    footerColumns: [],
    legalLinks: []
  }
};

describe('SFooter Snapshot', () => {
  // Define test scenarios with varying runtime configurations
  const scenarios: [string, { config: Record<string, unkown> }][] = [
    [
      'full config',
      {
        config: {
          app: { baseURL: '/' },
          public: {
            siteName: 'My Company',
            socialLinks: [
              {
                name: 'Twitter',
                href: 'https://twitter.com',
                icon: 'twitter-icon'
              },
              {
                name: 'Facebook',
                href: 'https://facebook.com',
                icon: 'facebook-icon'
              }
            ],
            brandLinks: [
              { name: 'Blog', href: '/blog' },
              { name: 'Careers', href: '/careers' }
            ],
            copyrightText: 'Â© 2023 My Company',
            address: '1234 Nuxt Street, Vue City',
            footerColumns: [
              {
                id: 1,
                title: 'Products',
                items: [
                  { slug: '/product-1', text: 'Product 1' },
                  { slug: '/product-2', text: 'Product 2' }
                ]
              },
              {
                id: 2,
                title: 'Support',
                items: [{ slug: '/support', text: 'Help Center' }]
              }
            ],
            legalLinks: [
              { text: 'Privacy Policy', slug: '/privacy' },
              { text: 'Terms of Service', slug: '/terms' }
            ]
          }
        }
      }
    ],
    [
      'minimal config with empty arrays',
      {
        config: {
          app: { baseURL: '/' },
          public: {
            siteName: 'Minimal Inc.',
            socialLinks: [],
            brandLinks: [],
            copyrightText: '',
            address: '',
            footerColumns: [],
            legalLinks: []
          }
        }
      }
    ]
  ];

  it.each(scenarios)(
    'renders %s correctly',
    async (desc: string, { config }) => {
      // Update mock runtime config for this scenario
      config_ = config;
      mockNuxtImport('useRuntimeConfig', () => () => config_);

      const html = await ComponentRender(`SFooter ${desc}`, {}, SFooter);
      expect(html).toMatchSnapshot();
    }
  );
});
