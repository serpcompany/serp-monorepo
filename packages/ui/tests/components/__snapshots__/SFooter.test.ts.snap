// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SFooter Snapshot > renders full config correctly 1`] = `
<footer class="
  bg-primary-800
  dark:bg-primary-950
  footer-wrapper
  mt-20
  w-full
">
  <div class="
    lg:py-0
    py-0
  ">
    <div class="
      py-30
      w-full
    ">
      <div class="
        max-w-[1248px]
        mx-auto
        px-4
      ">
        <nav class="
          xl:gap-8
          xl:grid
          xl:grid-cols-3
        ">
          <div class="
            mb-10
            xl:mb-0
          ">
            <div class="
              flex
              flex-col
              items-start
              space-x-2
            ">
              <span class="
                dark:text-primary-100
                font-bold
                pb-4
                text-4xl
                text-primary-100
              ">
                My Company
              </span>
              <p class="
                dark:text-primary-100
                font-bold
                text-primary-100
              ">
                Subscribe to the newsletter.
              </p>
              <p class="
                dark:text-primary-100
                text-primary-100
              ">
                Join ~trillions of informed readers.
              </p>
              <button
                class="
                  aria-disabled:bg-primary
                  aria-disabled:cursor-not-allowed
                  aria-disabled:opacity-75
                  bg-transparent
                  border
                  border-neutral-700
                  dark:bg-transparent
                  dark:border-neutral-700
                  dark:hover:bg-neutral-800
                  dark:text-white
                  disabled:bg-primary
                  disabled:cursor-not-allowed
                  disabled:opacity-75
                  focus-visible:outline-2
                  focus-visible:outline-offset-2
                  focus-visible:outline-primary
                  focus-visible:ring-(--ui-primary)
                  focus-visible:ring-2
                  focus:outline-hidden
                  font-semibold
                  gap-2
                  hover:bg-neutral-800
                  inline-flex
                  items-center
                  my-4
                  px-8
                  py-3
                  rounded-md
                  text-lg
                  text-white
                  transition-colors
                "
                type="button"
              >
                <!-- v-if -->
                <span class="truncate">
                  Subscribe
                </span>
                <!-- v-if -->
              </button>
              <!-- v-if -->
              <!-- teleport start -->
              <!-- teleport end -->
            </div>
          </div>
          <div class="
            auto-cols-fr
            flex
            flex-col
            gap-8
            grid-flow-col
            lg:grid
            xl:col-span-2
          ">
            <div>
              <h3 class="
                dark:text-primary-100
                font-semibold
                text-primary-100
                text-xl
              ">
                Products
              </h3>
              <ul class="
                mt-6
                space-y-4
              ">
                <li class="relative">
                  <a
                    class="
                      dark:hover:text-primary-200
                      dark:text-primary-100
                      flex
                      focus-visible:outline-primary
                      gap-1.5
                      group
                      hover:text-primary-200
                      items-center
                      text-primary-100
                      text-sm
                      transition-colors
                    "
                    href="/product-1"
                  >
                    
                  <!-- v-if --><span class="truncate">Product 1
                  <!-- v-if --></span>
                  </a>
                </li>
                <li class="relative">
                  <a
                    class="
                      dark:hover:text-primary-200
                      dark:text-primary-100
                      flex
                      focus-visible:outline-primary
                      gap-1.5
                      group
                      hover:text-primary-200
                      items-center
                      text-primary-100
                      text-sm
                      transition-colors
                    "
                    href="/product-2"
                  >
                    
                  <!-- v-if --><span class="truncate">Product 2
                  <!-- v-if --></span>
                  </a>
                </li>
              </ul>
            </div>
            <div>
              <h3 class="
                dark:text-primary-100
                font-semibold
                text-primary-100
                text-xl
              ">
                Support
              </h3>
              <ul class="
                mt-6
                space-y-4
              ">
                <li class="relative">
                  <a
                    class="
                      dark:hover:text-primary-200
                      dark:text-primary-100
                      flex
                      focus-visible:outline-primary
                      gap-1.5
                      group
                      hover:text-primary-200
                      items-center
                      text-primary-100
                      text-sm
                      transition-colors
                    "
                    href="/support"
                  >
                    
                  <!-- v-if --><span class="truncate">Help Center
                  <!-- v-if --></span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <!-- v-if -->
        </nav>
      </div>
    </div>
  </div>
  <div class="
    lg:flex
    lg:gap-x-3
    lg:items-center
    lg:justify-between
    lg:px-8
    lg:py-4
    max-w-(--ui-container)
    mx-auto
    px-4
    py-8
    sm:px-6
    w-full
  ">
    <div class="
      flex
      gap-x-1.5
      items-center
      justify-center
      lg:flex-1
      lg:justify-end
      lg:order-3
    "></div>
    <div class="
      flex
      items-center
      justify-center
      lg:mt-0
      lg:order-2
      lg:py-0
      mt-3
      py-0
    "></div>
    <div class="
      flex
      gap-x-1.5
      items-center
      justify-center
      lg:flex-1
      lg:justify-start
      lg:mt-0
      lg:order-1
      mt-3
    "></div>
  </div>
  <div class="
    lg:py-0
    py-0
  ">
    <div class="
      flex
      flex-col
      items-start
      justify-between
      max-w-[1248px]
      md:flex-row
      md:items-center
      mx-auto
      pb-8
      px-4
    ">
      <!-- <div class="flex flex-row items-center md:mb-0"> -->
      <!-- Social Links -->
      <div
        aria-label="Social Media Links"
        class="
          flex
          mb-8
          md:mb-0
        "
      >
        <a
          aria-label="Twitter"
          class="mr-6"
          href="https://twitter.com"
          rel="noopener noreferrer"
          target="_blank"
        ><span
          aria-hidden="true"
          class="
            dark:text-primary-100
            i-twitter-icon
            iconify
            size-5
            text-primary-100
          "
        ></span><span class="sr-only">Visit our Twitter page</span></a>
        <a
          aria-label="Facebook"
          class="mr-6"
          href="https://facebook.com"
          rel="noopener noreferrer"
          target="_blank"
        ><span
          aria-hidden="true"
          class="
            dark:text-primary-100
            i-facebook-icon
            iconify
            size-5
            text-primary-100
          "
        ></span><span class="sr-only">Visit our Facebook page</span></a>
      </div>
      <!-- </div> -->
      <div
        aria-label="Legal Navigation"
        class="
          flex
          flex-col
          md:flex-row
          sm:items-center
          text-sm
        "
      >
        <div class="
          flex
          flex-col
          flex-wrap
          gap-4
          justify-center
          md:justify-end
          sm:flex-row
        ">
          <a
            class="
              dark:text-primary-100
              text-primary-100
            "
            href="/privacy"
          >Privacy Policy</a>
          <a
            class="
              dark:text-primary-100
              text-primary-100
            "
            href="/terms"
          >Terms of Service</a>
        </div>
      </div>
    </div>
  </div>
</footer>
`;

exports[`SFooter Snapshot > renders minimal config with empty arrays correctly 1`] = `
<footer class="
  bg-primary-800
  dark:bg-primary-950
  footer-wrapper
  mt-20
  w-full
">
  <div class="
    lg:py-0
    py-0
  ">
    <div class="
      py-30
      w-full
    ">
      <div class="
        max-w-[1248px]
        mx-auto
        px-4
      ">
        <nav class="
          xl:gap-8
          xl:grid
          xl:grid-cols-3
        ">
          <div class="
            mb-10
            xl:mb-0
          ">
            <div class="
              flex
              flex-col
              items-start
              space-x-2
            ">
              <span class="
                dark:text-primary-100
                font-bold
                pb-4
                text-4xl
                text-primary-100
              ">
                Minimal Inc.
              </span>
              <p class="
                dark:text-primary-100
                font-bold
                text-primary-100
              ">
                Subscribe to the newsletter.
              </p>
              <p class="
                dark:text-primary-100
                text-primary-100
              ">
                Join ~trillions of informed readers.
              </p>
              <button
                class="
                  aria-disabled:bg-primary
                  aria-disabled:cursor-not-allowed
                  aria-disabled:opacity-75
                  bg-transparent
                  border
                  border-neutral-700
                  dark:bg-transparent
                  dark:border-neutral-700
                  dark:hover:bg-neutral-800
                  dark:text-white
                  disabled:bg-primary
                  disabled:cursor-not-allowed
                  disabled:opacity-75
                  focus-visible:outline-2
                  focus-visible:outline-offset-2
                  focus-visible:outline-primary
                  focus-visible:ring-(--ui-primary)
                  focus-visible:ring-2
                  focus:outline-hidden
                  font-semibold
                  gap-2
                  hover:bg-neutral-800
                  inline-flex
                  items-center
                  my-4
                  px-8
                  py-3
                  rounded-md
                  text-lg
                  text-white
                  transition-colors
                "
                type="button"
              >
                <!-- v-if -->
                <span class="truncate">
                  Subscribe
                </span>
                <!-- v-if -->
              </button>
              <!-- v-if -->
              <!-- teleport start -->
              <!-- teleport end -->
            </div>
          </div>
          <!-- v-if -->
          <!-- v-if -->
        </nav>
      </div>
    </div>
  </div>
  <div class="
    lg:flex
    lg:gap-x-3
    lg:items-center
    lg:justify-between
    lg:px-8
    lg:py-4
    max-w-(--ui-container)
    mx-auto
    px-4
    py-8
    sm:px-6
    w-full
  ">
    <div class="
      flex
      gap-x-1.5
      items-center
      justify-center
      lg:flex-1
      lg:justify-end
      lg:order-3
    "></div>
    <div class="
      flex
      items-center
      justify-center
      lg:mt-0
      lg:order-2
      lg:py-0
      mt-3
      py-0
    "></div>
    <div class="
      flex
      gap-x-1.5
      items-center
      justify-center
      lg:flex-1
      lg:justify-start
      lg:mt-0
      lg:order-1
      mt-3
    "></div>
  </div>
  <div class="
    lg:py-0
    py-0
  ">
    <div class="
      flex
      flex-col
      items-start
      justify-between
      max-w-[1248px]
      md:flex-row
      md:items-center
      mx-auto
      pb-8
      px-4
    ">
      <!-- <div class="flex flex-row items-center md:mb-0"> -->
      <!-- Social Links -->
      <div
        aria-label="Social Media Links"
        class="
          flex
          mb-8
          md:mb-0
        "
      ></div>
      <!-- </div> -->
      <div
        aria-label="Legal Navigation"
        class="
          flex
          flex-col
          md:flex-row
          sm:items-center
          text-sm
        "
      >
        <div class="
          flex
          flex-col
          flex-wrap
          gap-4
          justify-center
          md:justify-end
          sm:flex-row
        "></div>
      </div>
    </div>
  </div>
</footer>
`;

exports[`sFooter Snapshot > renders full config correctly 1`] = `
<footer class="
  bg-primary-800
  dark:bg-primary-950
  footer-wrapper
  mt-20
  w-full
">
  <div class="
    lg:py-0
    py-0
  ">
    <div class="
      py-30
      w-full
    ">
      <div class="
        max-w-[1248px]
        mx-auto
        px-4
      ">
        <nav class="
          xl:gap-8
          xl:grid
          xl:grid-cols-3
        ">
          <div class="
            mb-10
            xl:mb-0
          ">
            <div class="
              flex
              flex-col
              items-start
              space-x-2
            ">
              <span class="
                dark:text-primary-100
                font-bold
                pb-4
                text-4xl
                text-primary-100
              ">
                My Company
              </span>
              <p class="
                dark:text-primary-100
                font-bold
                text-primary-100
              ">
                Subscribe to the newsletter.
              </p>
              <p class="
                dark:text-primary-100
                text-primary-100
              ">
                Join ~trillions of informed readers.
              </p>
              <button
                class="
                  aria-disabled:bg-primary
                  aria-disabled:cursor-not-allowed
                  aria-disabled:opacity-75
                  bg-transparent
                  border
                  border-neutral-700
                  dark:bg-transparent
                  dark:border-neutral-700
                  dark:hover:bg-neutral-800
                  dark:text-white
                  disabled:bg-primary
                  disabled:cursor-not-allowed
                  disabled:opacity-75
                  focus-visible:outline-2
                  focus-visible:outline-offset-2
                  focus-visible:outline-primary
                  focus-visible:ring-(--ui-primary)
                  focus-visible:ring-2
                  focus:outline-hidden
                  font-semibold
                  gap-2
                  hover:bg-neutral-800
                  inline-flex
                  items-center
                  my-4
                  px-8
                  py-3
                  rounded-md
                  text-lg
                  text-white
                  transition-colors
                "
                type="button"
              >
                <!-- v-if -->
                <span class="truncate">
                  Subscribe
                </span>
                <!-- v-if -->
              </button>
              <!-- v-if -->
              <!-- teleport start -->
              <!-- teleport end -->
            </div>
          </div>
          <div class="
            auto-cols-fr
            flex
            flex-col
            gap-8
            grid-flow-col
            lg:grid
            xl:col-span-2
          ">
            <div>
              <h3 class="
                dark:text-primary-100
                font-semibold
                text-primary-100
                text-xl
              ">
                Products
              </h3>
              <ul class="
                mt-6
                space-y-4
              ">
                <li class="relative">
                  <a
                    class="
                      dark:hover:text-primary-200
                      dark:text-primary-100
                      flex
                      focus-visible:outline-primary
                      gap-1.5
                      group
                      hover:text-primary-200
                      items-center
                      text-primary-100
                      text-sm
                      transition-colors
                    "
                    href="/product-1"
                  >
                    
                  <!-- v-if --><span class="truncate">Product 1
                  <!-- v-if --></span>
                  </a>
                </li>
                <li class="relative">
                  <a
                    class="
                      dark:hover:text-primary-200
                      dark:text-primary-100
                      flex
                      focus-visible:outline-primary
                      gap-1.5
                      group
                      hover:text-primary-200
                      items-center
                      text-primary-100
                      text-sm
                      transition-colors
                    "
                    href="/product-2"
                  >
                    
                  <!-- v-if --><span class="truncate">Product 2
                  <!-- v-if --></span>
                  </a>
                </li>
              </ul>
            </div>
            <div>
              <h3 class="
                dark:text-primary-100
                font-semibold
                text-primary-100
                text-xl
              ">
                Support
              </h3>
              <ul class="
                mt-6
                space-y-4
              ">
                <li class="relative">
                  <a
                    class="
                      dark:hover:text-primary-200
                      dark:text-primary-100
                      flex
                      focus-visible:outline-primary
                      gap-1.5
                      group
                      hover:text-primary-200
                      items-center
                      text-primary-100
                      text-sm
                      transition-colors
                    "
                    href="/support"
                  >
                    
                  <!-- v-if --><span class="truncate">Help Center
                  <!-- v-if --></span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <!-- v-if -->
        </nav>
      </div>
    </div>
  </div>
  <div class="
    lg:flex
    lg:gap-x-3
    lg:items-center
    lg:justify-between
    lg:px-8
    lg:py-4
    max-w-(--ui-container)
    mx-auto
    px-4
    py-8
    sm:px-6
    w-full
  ">
    <div class="
      flex
      gap-x-1.5
      items-center
      justify-center
      lg:flex-1
      lg:justify-end
      lg:order-3
    "></div>
    <div class="
      flex
      items-center
      justify-center
      lg:mt-0
      lg:order-2
      lg:py-0
      mt-3
      py-0
    "></div>
    <div class="
      flex
      gap-x-1.5
      items-center
      justify-center
      lg:flex-1
      lg:justify-start
      lg:mt-0
      lg:order-1
      mt-3
    "></div>
  </div>
  <div class="
    lg:py-0
    py-0
  ">
    <div class="
      flex
      flex-col
      items-start
      justify-between
      max-w-[1248px]
      md:flex-row
      md:items-center
      mx-auto
      pb-8
      px-4
    ">
      <!-- <div class="flex flex-row items-center md:mb-0"> -->
      <!-- Social Links -->
      <div
        aria-label="Social Media Links"
        class="
          flex
          mb-8
          md:mb-0
        "
      >
        <a
          aria-label="Twitter"
          class="mr-6"
          href="https://twitter.com"
          rel="noopener noreferrer"
          target="_blank"
        ><span
          aria-hidden="true"
          class="
            dark:text-primary-100
            i-twitter-icon
            iconify
            size-5
            text-primary-100
          "
        ></span><span class="sr-only">Visit our Twitter page</span></a>
        <a
          aria-label="Facebook"
          class="mr-6"
          href="https://facebook.com"
          rel="noopener noreferrer"
          target="_blank"
        ><span
          aria-hidden="true"
          class="
            dark:text-primary-100
            i-facebook-icon
            iconify
            size-5
            text-primary-100
          "
        ></span><span class="sr-only">Visit our Facebook page</span></a>
      </div>
      <!-- </div> -->
      <div
        aria-label="Legal Navigation"
        class="
          flex
          flex-col
          md:flex-row
          sm:items-center
          text-sm
        "
      >
        <div class="
          flex
          flex-col
          flex-wrap
          gap-4
          justify-center
          md:justify-end
          sm:flex-row
        ">
          <a
            class="
              dark:text-primary-100
              text-primary-100
            "
            href="/privacy"
          >Privacy Policy</a>
          <a
            class="
              dark:text-primary-100
              text-primary-100
            "
            href="/terms"
          >Terms of Service</a>
        </div>
      </div>
    </div>
  </div>
</footer>
`;

exports[`sFooter Snapshot > renders minimal config with empty arrays correctly 1`] = `
<footer class="
  bg-primary-800
  dark:bg-primary-950
  footer-wrapper
  mt-20
  w-full
">
  <div class="
    lg:py-0
    py-0
  ">
    <div class="
      py-30
      w-full
    ">
      <div class="
        max-w-[1248px]
        mx-auto
        px-4
      ">
        <nav class="
          xl:gap-8
          xl:grid
          xl:grid-cols-3
        ">
          <div class="
            mb-10
            xl:mb-0
          ">
            <div class="
              flex
              flex-col
              items-start
              space-x-2
            ">
              <span class="
                dark:text-primary-100
                font-bold
                pb-4
                text-4xl
                text-primary-100
              ">
                Minimal Inc.
              </span>
              <p class="
                dark:text-primary-100
                font-bold
                text-primary-100
              ">
                Subscribe to the newsletter.
              </p>
              <p class="
                dark:text-primary-100
                text-primary-100
              ">
                Join ~trillions of informed readers.
              </p>
              <button
                class="
                  aria-disabled:bg-primary
                  aria-disabled:cursor-not-allowed
                  aria-disabled:opacity-75
                  bg-transparent
                  border
                  border-neutral-700
                  dark:bg-transparent
                  dark:border-neutral-700
                  dark:hover:bg-neutral-800
                  dark:text-white
                  disabled:bg-primary
                  disabled:cursor-not-allowed
                  disabled:opacity-75
                  focus-visible:outline-2
                  focus-visible:outline-offset-2
                  focus-visible:outline-primary
                  focus-visible:ring-(--ui-primary)
                  focus-visible:ring-2
                  focus:outline-hidden
                  font-semibold
                  gap-2
                  hover:bg-neutral-800
                  inline-flex
                  items-center
                  my-4
                  px-8
                  py-3
                  rounded-md
                  text-lg
                  text-white
                  transition-colors
                "
                type="button"
              >
                <!-- v-if -->
                <span class="truncate">
                  Subscribe
                </span>
                <!-- v-if -->
              </button>
              <!-- v-if -->
              <!-- teleport start -->
              <!-- teleport end -->
            </div>
          </div>
          <!-- v-if -->
          <!-- v-if -->
        </nav>
      </div>
    </div>
  </div>
  <div class="
    lg:flex
    lg:gap-x-3
    lg:items-center
    lg:justify-between
    lg:px-8
    lg:py-4
    max-w-(--ui-container)
    mx-auto
    px-4
    py-8
    sm:px-6
    w-full
  ">
    <div class="
      flex
      gap-x-1.5
      items-center
      justify-center
      lg:flex-1
      lg:justify-end
      lg:order-3
    "></div>
    <div class="
      flex
      items-center
      justify-center
      lg:mt-0
      lg:order-2
      lg:py-0
      mt-3
      py-0
    "></div>
    <div class="
      flex
      gap-x-1.5
      items-center
      justify-center
      lg:flex-1
      lg:justify-start
      lg:mt-0
      lg:order-1
      mt-3
    "></div>
  </div>
  <div class="
    lg:py-0
    py-0
  ">
    <div class="
      flex
      flex-col
      items-start
      justify-between
      max-w-[1248px]
      md:flex-row
      md:items-center
      mx-auto
      pb-8
      px-4
    ">
      <!-- <div class="flex flex-row items-center md:mb-0"> -->
      <!-- Social Links -->
      <div
        aria-label="Social Media Links"
        class="
          flex
          mb-8
          md:mb-0
        "
      ></div>
      <!-- </div> -->
      <div
        aria-label="Legal Navigation"
        class="
          flex
          flex-col
          md:flex-row
          sm:items-center
          text-sm
        "
      >
        <div class="
          flex
          flex-col
          flex-wrap
          gap-4
          justify-center
          md:justify-end
          sm:flex-row
        "></div>
      </div>
    </div>
  </div>
</footer>
`;
